### PR TITLE
Make scheduled talks accessible to everyone in the API

### PR DESCRIPTION
**Pull request by kayzhu**
~~This is achieved by overwriting TalkDao's `get()` and `index()` methods to check if the talk has `schedule_slot` or not. Probably not the best way to do this though. I will think of a cleaner fix.~~

Minor fix for views.

**UPDATE**: Don't merge yet.. I will fix the tests first.

**UPDATE 2**: With @dianaclarke 's and  @wolever 's help, this new change should pass all the tests and also make scheduled talks available to the public.  Scheduled talks are now made accessible by changing `_query` method in `TalkDao` such that only talks with `schedule_slot` or belong to the current user will be accessible for non-admins.

**UPDATE 3**: Fixed "title" problems for various templates, also fixed form styling problems for password reset related code.

**UPDATE 4**: Now "Edit" button on talk page only shows up when the user is admin or owner of the talk.

### CODE REVIEW

File: pyconca/dao/talk_dao.py [hunk 1]
```diff
@@ -14,7 +15,13 @@ def _query(self):
...
-            raise Exception  # TODO
+        if self.authenticated_user:
+            return query.filter(or_(
+                (Talk.schedule_slot != None),
+                (Talk.owner_id == self.authenticated_user.id)
+            ))
+        return query.filter(Talk.schedule_slot != None)
+
+    def get_owner(self, talk_id):
+        talk = self._query().filter_by(id=talk_id).first()
```
Comment (eloraburns) [inline pyconca/dao/talk_dao.py:26]: Would `self._query().get(talk_id)` work here, too?

**Comment by eloraburns** [comment]
Sorry I took so long to look at this. :(

It was a bit tricky to see what was going on due to the password reset code being inlined… Or is that a strange bug of github?

File: pyconca/dao/talk_dao.py (hunk 1 above)
Comment (kayzhu) [inline pyconca/dao/talk_dao.py:26]: Ahh you are absolutely right! I will make that correction now :D

**Comment by kayzhu** [comment]
Ah not at all :) 

Yes it was basically two main pull requests consolidated into one.. I think all the `.mako` file changes were the "other" part of the pull requests. The rest is pretty much all about making the talk API public.

**Pull request merged**